contrib/platform/src/com/sun/jna/platform/win32/Kernel32Util.java

```diff
@@ -678,4 +678,89 @@ public static final String extractVolumeGUID(String volumeGUIDPath) {
         
         return volumeGUIDPath.substring(VOLUME_GUID_PATH_PREFIX.length(), volumeGUIDPath.length() - VOLUME_GUID_PATH_SUFFIX.length());
     }
+
+    /**
+     * Gets the specified resource out of the specified executable file
+     * 
+     * @param path
+     *            The path to the executable file
+     * @param type
+     *            The type of the resource (either a type name or type ID is
+     *            allowed)
+     * @param name
+     *            The name or ID of the resource
+     * @return The resource bytes, or null if no such resource exists.
+     * @throws IllegalStateException if the call to LockResource fails
+     */
+    public static byte[] getResource(String path, String type, String name) {
+        HMODULE target = Kernel32.INSTANCE.LoadLibraryEx(path, null, Kernel32.LOAD_LIBRARY_AS_DATAFILE);
+
+        if (target == null) {
+            throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+        }
+        
+        Win32Exception err = null;
+        Pointer start = null;
+        int length = 0;
+        try {
+            Pointer t = null;
+            try {
+                t = new Pointer(Long.parseLong(type));
+            } catch (NumberFormatException e) {
+                t = new Memory(Native.WCHAR_SIZE * (type.length() + 1));
+                t.setWideString(0, type);
+            }
+
+            Pointer n = null;
+            try {
+                n = new Pointer(Long.parseLong(name));
+            } catch (NumberFormatException e) {
+                n = new Memory(Native.WCHAR_SIZE * (name.length() + 1));
+                n.setWideString(0, name);
+            }
+
+            HRSRC hrsrc = Kernel32.INSTANCE.FindResource(target, n, t);
+            if (hrsrc == null) {
+                throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+            }
+
+            // according to MSDN, on 32 bit Windows or newer, calling FreeResource() is not necessary - and in fact does nothing but return false.
+            HANDLE loaded = Kernel32.INSTANCE.LoadResource(target, hrsrc);
+            if (loaded == null) {
+                throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+            }
+            
+            length = Kernel32.INSTANCE.SizeofResource(target, hrsrc);
+            if (length == 0) {
+                throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
+            }
+            
+            // MSDN: It is not necessary to unlock resources because the system automatically deletes them when the process that created them terminates.
+            // MSDN does not say that LockResource sets GetLastError
+            start = Kernel32.INSTANCE.LockResource(loaded);
+            if (start == null) {
+                throw new IllegalStateException("LockResource returned null.");
+            }
+        } catch (Win32Exception we) {
+            err = we;
```

Comment:
Why do you need to store `err`, why not remove the `catch` and just let it throw?